alica_tests/src/test/test_blackboard.cpp

```diff
@@ -50,18 +52,99 @@ class TestBlackboard : public AlicaTestFixture
+            return std::any_cast<SrcType>(targetBB.impl().get<TargetType>("targetKey")) == std::any_cast<SrcType>(targetValue);
         } else {
-            return targetBB._impl.get<TargetType>("targetKey") == targetValue;
+            return targetBB.impl().get<TargetType>("targetKey") == targetValue;
         }
     }
 
-    alica::internal::BlackboardImpl& getBlackboardImpl(alica::Blackboard& bb) { return bb._impl; }
+    alica::internal::BlackboardImpl& getBlackboardImpl(alica::Blackboard& bb) { return bb.impl(); }
 };
 
+TEST_F(SingleAgentTestFixture, testValueMappingBehaviours)
+{
+    // Checks if value mapping for behaviours succeeds
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("BlackboardTestPlan"));
+    auto plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("BlackboardTestPlan", "ChooseBlackboardTestState", "ValueMappingBehaviourTestState")) << _tc->getLastFailure();
+    plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    // Step until the plan succeeds
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->isSuccess(plan));
+}
+
+TEST_F(SingleAgentTestFixture, testValueMappingConditions)
+{
+    // Checks if value mapping for conditions succeeds
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("BlackboardTestPlan"));
+    auto plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("BlackboardTestPlan", "ChooseBlackboardTestState", "ValueMappingConditionTestState")) << _tc->getLastFailure();
+    plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    // Step until the plan succeeds
+    STEP_UNTIL_ASSERT_TRUE(_tc, _tc->isSuccess(plan));
+}
+
+TEST_F(SingleAgentTestFixture, testValueMappingPlans)
+{
+    // Checks if value mapping for conditions succeeds
+
+    // Transition to the plan corresponding to this test case
+    ASSERT_TRUE(_tc->setTransitionCond("TestMasterPlan", "ChooseTestState", "BlackboardTestState")) << _tc->getLastFailure();
+    STEP_UNTIL(_tc, _tc->getActivePlan("BlackboardTestPlan"));
+    auto plan = _tc->getActivePlan("BlackboardTestPlan");
+    ASSERT_NE(plan, nullptr) << _tc->getLastFailure();
+
+    ASSERT_TRUE(_tc->setTransitionCond("BlackboardTestPlan", "ChooseBlackboardTestState", "ValueMappingPlanTestState")) << _tc->getLastFailure();
+    plan = _tc->getActivePlan("BlackboardTestPlan");
```

Comment:
not required. Instead, to check if the mapping worked replace the AlwaysTrueCondition with a condition that checks if there is no error is set on the global blackboard. Basically the value mapping behs, plans & conds set the failure on the global blackboard. If not set that means there was no error.